checker/src/org/checkerframework/checker/units/UnitsAnnotatedTypeFactory.java

```diff
@@ -189,18 +190,19 @@ public AnnotationMirror aliasedAnnotation(AnnotationMirror a) {
      * @param qual The qualifier to investigate.
      */
     private void addUnitsRelations(Class<? extends Annotation> qual) {
+        // get the annotation mirror
         AnnotationMirror am = AnnotationUtils.fromClass(elements, qual);
-
+        // get all the meta annotations of the class
         for (AnnotationMirror ama : am.getAnnotationType().asElement().getAnnotationMirrors() ) {
-            if (ama.getAnnotationType().toString().equals(UnitsRelations.class.getCanonicalName())) {
+            // if there's a UnitsRelations meta annotation
+            if (ama.getAnnotationType().toString().equals(org.checkerframework.checker.units.qual.UnitsRelations.class.getCanonicalName())) {
```

Comment:
I think this was the most important change in this pull request. You can make this even clearer by extracting the UnitsRelations into a Class<? extends Annotation> variable and then comparing.